src/index.ts

```diff
@@ -138,3 +139,54 @@ export function useLiveAnchorAccount<
     slotUpdated,
   };
 }
+
+function fetcher<I extends Idl, A extends keyof IdlAccounts<I>>(
+  program: Program<I>,
+) {
+  return function (method: A, ...params: [Address]): IdlAccounts<I>[A] {
+    const address = params[0];
+    return program.account[method].fetch(address) as IdlAccounts<I>[A];
+  };
+}
+
+export interface UseSWRAnchorAccountResult<
+  I extends Idl,
+  A extends keyof IdlAccounts<I>,
+> {
+  loading: boolean;
+  account?: IdlAccounts<I>[A];
+  mutate: KeyedMutator<IdlAccounts<I>[A]>;
+  error?: string;
+  isValidating: boolean;
+}
+
+export function useSWRAnchorAccount<
+  I extends Idl,
+  A extends keyof IdlAccounts<I>,
+>(
+  program: Program<I> | null | undefined,
+  accountType: A,
+  address: Address | null | undefined,
+  swrOptions?: SWRConfiguration,
+): UseSWRAnchorAccountResult<I, A> {
+  const {
+    data: account,
+    error,
+    mutate,
+    isValidating,
+  } = useSWR<IdlAccounts<I>[A]>(
+    program && address && [accountType, address],
+    fetcher(program!),
+    swrOptions,
+  );
+
+  const loading = !account;
+
+  return {
+    loading,
+    account: account as IdlAccounts<I>[A],
```

Comment:
nit:

```suggestion
    account,
```

The type check seems to pass for me without `as`, is it failing for you?

Reply:
Good catch, I added that before I was able to specify a type of `account`